docs/source/parameters.rst

```diff
@@ -0,0 +1,373 @@
+
+Defined parameters can be used in steps directly:
+
+.. code-block:: yaml
+   :linenos:
+
+   - name: run-lulesh
+     description: Run LULESH.
+     run:
+       cmd: |
+           $(LULESH)/lulesh2.0 -s $(SIZE) -i $(ITERATIONS) -p > $(outfile)
+       depends: [make-lulesh]
+
+Even though this is defined in Maestro as a single step, Maestro will automatically run this step with each parameter combinations. This makes it very easy to setup studies and apply parameters to be run.
+
+.. note::
+
+   Maestro will only use parameters if they've been defined in at least one step
+
+In addition to direct access to parameter values, parameter labels can also be used in steps:
+
+.. code-block:: yaml
+   :linenos:
+
+   - name: run-lulesh
+     description: Run LULESH.
+     run:
+       cmd: |
+           echo "Running case: $(SIZE.label), $(ITERATIONS.label)"
+           $(LULESH)/lulesh2.0 -s $(SIZE) -i $(ITERATIONS) -p > $(outfile)
+       depends: [make-lulesh]
+
+.. note
+   add example with using the label in steps too
+
+What can be Parameterized in Maestro?
+=====================================
+
+A common use case for Maestro is to use the parameter block to specify values to iterate over for a simulation parameter study; however, Maestro does not make any assumptions about what these values are. This makes the use of Maestro's parameter block very flexible. For example, Maestro does not require the parameter variations to be numeric.
+
+.. code-block:: yaml
+   :linenos:
+   :caption: study.yaml
+
+   study:
+    - name: run-simulation
+      description: Run a simulation.
+      run:
+        cmd: |
+            $(CODE) -in $(SPECROOT)/$(INPUT)
+      depends: []
+
+   global.parameters:
+       INPUT:
+           values  : [input1.in, input2.in, input3.in]
+           label   : INPUT.%%
+
+The above example highlights a partial study spec that defines a parameter block of simulation inputs that will be varied when the study runs. The **run-simulation** step  will run three times, once for each defined input file.
+
+.. code-block:: yaml
+   :linenos:
+   :caption: study.yaml
+
+   study:
+       - name: run-simulation
+         description: Run a simulation.
+         run:
+           cmd: |
+               $(CODE_PATH)/$(VERSION)/code.exe -in $(SPECROOT)/$(INPUT)
+         depends: []
+
+   global.parameters:
+       INPUT:
+           values  : [input1.in, input2.in, input3.in, input1.in, input2.in, input3.in]
+           label   : INPUT.%%
+       VERSION:
+           values  : [4.0.0, 4.0.0, 4.0.0, 5.0.0, 5.0.0, 5.0.0]
+           label   : VERSION.%%
+
+This example parameterizes the inputs and the version of the code being run.  Maestro will run each input with the different code version.  The above example assumes that all the code versions share a base path, **$(CODE_PATH)** which is inserted via the token replacment mechanism to yeild the full paths (e.g. /usr/gapps/code/4.0.0/code.exe).
```

Comment:
I feel like paths and references to YAML specific components: `$(CODE_PATH)` is more easily visible as a YAML component.

Reply:
meaning revert the in-text mentions of monikers to be ``$(CODE_PATH)`` instead of the bolded format here?

Reply:
I think that would look better. What do you think?